src/common/src/cpu.rs

```diff
@@ -608,6 +620,56 @@ pub fn clean_and_invalidate_data_cache(virtual_address: usize) {
     unsafe { asm!("DC CIVAC, {:x}", in(reg) virtual_address) };
 }
 
+pub fn clean_data_cache_all() {
+    dsb();
+    let clidr_el1: u64;
+    unsafe { asm!("mrs {:x}, clidr_el1", out(reg) clidr_el1) };
+    let loc = (clidr_el1 & CLIDR_EL1_LOC) >> CLIDR_EL1_LOC_BITS_OFFSET;
+    for cache_level in 0..loc {
+        let cache_type = (clidr_el1 >> (3 * cache_level)) & 0b111;
+        let ccsidr_el1: u64;
+
+        if cache_type <= 1 {
+            /* Data Cache is not available */
+            continue;
+        }
+        unsafe {
+            asm!("
+                    msr csselr_el1, {:x}
+                    isb
+                    mrs {:x}, ccsidr_el1
+                ", in(reg) cache_level << 1, out(reg) ccsidr_el1)
+        };
+
+        let line_size =
+            ((ccsidr_el1 & CCSIDR_EL1_LINE_SIZE) >> CCSIDR_EL1_LINE_SIZE_BITS_OFFSET) + 4;
+        let associativity =
+            ((ccsidr_el1 & CCSIDR_EL1_ASSOCIATIVITY) >> CCSIDR_EL1_ASSOCIATIVITY_BITS_OFFSET) + 1;
+        let num_sets = ((ccsidr_el1 & CCSIDR_EL1_NUM_SETS) >> CCSIDR_EL1_NUM_SETS_BITS_OFFSET) + 1;
+        let set_way_a = (associativity as u32 - 1).leading_zeros();
+
+        for set in 0..num_sets {
+            for way in 0..associativity {
+                /* C5.3.13 DC CISW, Data or unified Cache line Clean and Invalidate by Set/Way (ARM DDI 0487G.a ID011921)
```

Comment:
Nit: This information looks a little bit old. The latest manual version is ARM DDI 0487J.a (ID042523), and the section number of DC CISW is C5.3.15.